Desktop/AddGoalWindow.cs

```diff
@@ -110,41 +120,53 @@ private void goalAmount_KeyPress(object sender, KeyPressEventArgs e)
             }
         }
 
-        public void ValidateFields(string amount, string details)
+        public void ValidateFields(string amount, string details, string name)
         {
             if (String.IsNullOrWhiteSpace(amount))
             {
-                goalMoney.BackColor = Color.Red;
+                  goalMoney.BackColor = Color.Red;
             }
 
             if (String.IsNullOrWhiteSpace(details))
             {
-                goalMoney.BackColor = Color.Red;
+                descriptionBox.BackColor = Color.Red;
+            }
+
+            if (String.IsNullOrWhiteSpace(name))
+            {
+                goalNameBox.BackColor = Color.Red;
             }
```

Comment:
Consider putting status colours into an Enum

Reply:
I don't think Enum is useful when using only one color

Reply:
Oh, I missed that, you are right!